dxp/layout_rendering.md

```diff
@@ -0,0 +1,173 @@
+-   `modules/apps/archived/portal-security-wedeploy-auth-web/src/main/java/com/liferay/portal/security/wedeploy/auth/web/internal/struts/WeDeployAccessTokenStrutsAction.java`
+-   `modules/apps/archived/portal-security-wedeploy-auth-web/src/main/java/com/liferay/portal/security/wedeploy/auth/web/internal/struts/WeDeployAuthorizeStrutsAction.java`
+-   `modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/struts/ExportLayoutPageTemplateEntriesStrutsAction.java`
+
+As with `portal-impl`'s `Action`s, we should expect no more `StrutsAction`s to appear in DXP's codebase.
+
+## `LayoutAction`
+
+This is the specific `Action` responsible for processing all requests to `/c/portal/layout`.
+
+## `TilesUtil`
+
+This is the support for the legacy Struts Tiles. It has a `loadDefinitions()` method which is invoked by `MainServlet`, upon initialization, to load `TilesUtil.DEFINITIONS` to be used by `PortalRequestProcessor` to map Struts forwards usings the `_definitions` map.
+
+The Tiles defininitions are loaded from the `/WEB-INF/tiles-defs.xml` file.
+
+## `portal.jsp`
+
+This is a JSP file that checks the value of the `pop_up` attribute in the active Tiles definition and then invokes `portal_pop_up.jsp` or `portal_normal.jsp` depending on its value. This way, the content being rendered is drawn as a popup dialog or a regular HTML page.
+
+It can also call `portal_pop_up.jsp` if the current request contains a `WIDGET` attribute set to `true`. This happens when the `WidgetServlet` services any URL starting with `/widget/...`.
+
+## `WidgetServlet`
+
+This servlet listens to `/widget/...` URLs and reforwards whatever comes after `/widget/` to be processed again but before it sets a `WIDGET` attribute to `true` in the request, so that the content is rendered using the `portal_pop_up.jsp` template instead of the `portal_normal.jsp` one.
+
+> Note, however, that not every URL can be routed through `/widget/...` and expect it to be shown correctly inside a popup.
+
+> This was used in the past by the `Liferay.Widget` API, but it is now deprecated and not used in DXP.
+
+## `ThemeImpl`, `portal_pop_up.jsp` and `portal_normal.jsp`
+
+Inside `portal-web/docroot/html/common/themes` there used to be two core JSP files to render content. However, with the advent of themes, these two files were diverted to templates inside the active theme. So, for example, in a vanilla DXP installation they get routed to:
+
+-   `portal_pop_up.jsp`: `portal_pop_up.ftl`
+-   `portal_normal.jsp`: `portal_normal.ftl`
+
+Both FTL files inside `classic-theme` or `admin-theme` (whichever one is active for the request).
+
+This mapping happens in [`ThemeImpl.getResourcePath()`](https://github.com/liferay/liferay-portal/blob/02716cc55e569eb315783aafb678d73fd90ad971/portal-impl/src/com/liferay/portal/model/impl/ThemeImpl.java#L688).
+
+# How all this fits together
```

Comment:
The anticipation and not knowing is killing me 😂

Reply:
I'm finally removing that section because I reworded the previous ones.

![shrugh](https://media.giphy.com/media/3o7TKHKjrDyqphX9Cg/giphy.gif)